plugins/woocommerce/src/Admin/Features/ProductBlockEditor/RedirectionController.php

```diff
@@ -63,7 +64,15 @@ protected function is_legacy_edit_screen(): bool {
 	protected function is_product_supported( $product_id ): bool {
 		$product = $product_id ? wc_get_product( $product_id ) : null;
 		$digital_product = $product->is_downloadable() || $product->is_virtual();
-		return $product && in_array( $product->get_type(), $this->supported_post_types, true ) && ! $digital_product;
+
+		if ( $product && in_array( $product->get_type(), $this->supported_post_types, true ) ) {
+			if ( Features::is_enabled( 'product-virtual-downloadable' ) ) {
+				return $digital_product;
```

Comment:
This doesn't work, as this will cause the `redirect` to only work for digital products and nothing else.
I think you just want to return `true` here? or do an additional check against `virtual`.

Reply:
Right!!! done -> https://github.com/woocommerce/woocommerce/pull/40731/commits/48e7f21b180faec0463074d45e9ebfb6be692469